### ISSUE TITLE
Incorrect repr() of certain strategies when running with xdis 3.2.0 and PyPy

### ISSUE THREAD

**Issue by alexwlchan** [HypothesisWorks/hypothesis-python#382]
This continues the discussion from #379.

If you look at the [Travis run](https://travis-ci.org/HypothesisWorks/hypothesis-python/builds/171772822) with unpinned dependencies, that installed uncompyle6&nbsp;v2.9.3 and xdis&nbsp;v3.2.0. There are two failures on PyPy (macOS and Linux) around repr’s, specifically:

```
>>>> from hypothesis import strategies as st
>>>> from hypothesis.internal.reflection import get_pretty_function_description

>>>> get_pretty_function_description(lambda right: [].map(lambda length:()))
'lambda right: [].None(lambda length: ())'

>>>> repr(st.integers().flatmap(lambda x: st.booleans()))
'integers().flatmap(lambda x: st.None())'
```

I can reproduce these locally.

<details>
<summary>

Environment config</summary>

``` console
$ pip freeze
cffi==1.3.0
enum34==1.1.6
greenlet==0.4.9
hypothesis==3.5.3
readline==6.2.4.1
spark-parser==1.4.0
uncompyle6==2.9.3
wheel==0.24.0
xdis==3.2.0

$ python --version
Python 2.7.10 (850edf14b2c75573720f59e95767335fb1affe55, Oct 30 2015, 00:18:28)
[PyPy 4.0.0 with GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.1.76)]
```

</details>

These failures do not occur if you pin to uncompyle6 v2.9.2 and xdis v3.1.0, as verified by Travis.

Playing with `git bisect` suggests this bug was introduced in https://github.com/rocky/python-xdis/commit/ea8649d9b2b168c27f8409508d29cbb90f345e1e, but I don’t know enough about how opcodes work to identify which part of that commit is causing us grief, and I don’t fancy unpicking the diff by hand tonight.

I’ll file an upstream bug, although I don’t fully understand what‘s going on here. More investigation required.

**Comment by alexwlchan** [comment]
We now have xdis 3.2.1 which should fix this bug; I’ve confirmed the fix locally, and triggered another Travis run in #381. Assuming that works, our dependency issues should be over (for now, at least).

**Comment by alexwlchan** [comment]
We have a clean CI run if you update to xdis 3.2.1, and I can no longer reproduce this issue locally.

Fix is to update to xdis 3.2.1, no changes required to Hypothesis.
